clang/docs/HIPSupport.rst

```diff
@@ -210,6 +210,88 @@ Host Code Compilation
 - These relocatable objects are then linked together.
 - Host code within a TU can call host functions and launch kernels from another TU.
 
+HIP Fat Binary Registration and Unregistration
+==============================================
+
+When compiling HIP for AMD GPUs, Clang embeds device code into HIP "fat
+binaries" and generates host-side helper functions that register these
+fat binaries with the HIP runtime at program start and unregister them at
+program exit. In non-RDC mode (``-fno-gpu-rdc``), each compilation unit
```

Comment:
I'd add some details on what is that registration is used for. I.e. to associate host-side addresses with GPU-side entities. E.g. when we call a host-side kernel stub, runtime needs to know which kernel symbol on the GPU side we intend to call.